tests/kernel/io_uring.pm

```diff
@@ -0,0 +1,132 @@
+# SUSE's openQA tests
+#
+# Copyright 2023 SUSE LLC
+# SPDX-License-Identifier: FSFAP
+#
+# Summary: Executes liburing testing suite
+# Maintainer: Kernel QE <kernel-qa@suse.de>
+# More documentation is at the bottom
+
+use base 'opensusebasetest';
+use strict;
+use warnings;
+use testapi;
+use serial_terminal 'select_serial_terminal';
+use utils;
+use LTP::WhiteList;
+
+sub run {
+    my $self = shift;
+
+    select_serial_terminal;
+
+    my $repository = get_var('LIBURING_REPO', 'https://github.com/axboe/liburing.git');
+    my $timeout = get_var('LIBURING_TIMEOUT', 1800);
+    my $version = get_var('LIBURING_VERSION', '');
+    my $exclude = get_var('LIBURING_EXCLUDE', '');
+    my $pkgs = "git-core";
+    my @lines;
+    my $out;
+
+    # check if liburing2 is installed and eventually install it
+    $out = script_output("rpm -q liburing2");
+    if ($out =~ /not installed/) {
+        $pkgs .= " liburing2";
+    }
+
+    # install dependences
+    zypper_call("in -t pattern devel_basis");
+    zypper_call("in $pkgs");
+
+    # select latest liburing version which is supported by the system
+    if ($version eq '') {
+        $out = script_output('rpm -q --qf "%{Version}\n" liburing2 | sort -nr | head -1');
+        $version = "liburing-$out";
+    }
+
+    # download and compile tests
+    assert_script_run("git clone --no-single-branch $repository");
+    assert_script_run("git -C liburing checkout $version");
+    assert_script_run("cd liburing");
+    assert_script_run("./configure");
+    assert_script_run("make -C src");
+    assert_script_run("make -C test");
+
+    # run tests executables
+    if ($exclude ne '') {
+        assert_script_run("echo TEST_EXCLUDE=\"$exclude\" > test/config.local");
+
+        record_info(
+            "Exclude",
+            "Excluding tests: $exclude",
+            result => 'softfail'
+        );
+    }
+
+    $out = script_output(
+        "make -C test runtests",
+        timeout => $timeout,
+        proceed_on_failure => 1
+    );
+
+    # search for timed out tests
+    @lines = $out =~ /Tests timed out \(d+\):.*/mg;
+    if (@lines) {
+        record_info(
+            "Timeout",
+            $lines[0],
+            result => 'softfail'
+        );
+    }
+
+    # search for failed tests and known issues
+    @lines = $out =~ /Tests failed \(\d+\):.*/mg;
+    if (@lines) {
+        my $whitelist = LTP::WhiteList->new();
+        my $environment = {
+            product => get_var('DISTRI') . ':' . get_var('VERSION'),
+            revision => get_var('BUILD'),
+            flavor => get_var('FLAVOR'),
+            arch => get_var('ARCH'),
+            backend => get_var('BACKEND'),
+            kernel => '',
```

Comment:
You should also set `kernel => script_output('uname -r')` in order to allow whitelisting specific kernels only. `ltp_version` could be set to the io_uring test version if we plan to build a stable test package.

It'd also be a good idea to collect this info before you start testing because when the tests fail, the system may be dead and you won't be able to query the kernel version anymore.

Reply:
BTW, LTP::Whitelist supports skip rules which should be preferred over `LIBURING_EXCLUDE`. That's another reason to collect env info before running the tests.